abi/encoding_test.go

```diff
@@ -183,6 +183,20 @@ func TestEncoding(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
Please add also a test for the float.

Reply:
sorry I pushed the PR before finishing the tests properly. I now added tests for float64 and addresses and also for the polymorphism issue.

I had to create a new test function because the previous test was not asserting what I needed. They were using `testEncodeDecode` but what I expect is to pass an input such as `float64` and check that the result is `int64, int32, uint32, etc...`. For addresses as strings I'm expecting to pass a `string` and receive an `web3.Address` back.